modules/mobile/app/controllers/mobile/v0/folders_controller.rb

```diff
@@ -1,26 +1,27 @@
 # frozen_string_literal: true
 
+require_relative '../concerns/json_api_pagination_links'
+
 module Mobile
   module V0
     class FoldersController < MessagingController
+      include Mobile::Concerns::JsonApiPaginationLinks
+
       def index
         resource = client.get_folders(@current_user.uuid, use_cache?)
+        links = pagination_links(resource)
```

Comment:
It looks like this `pagination_links` adds all the proper links (first, last, previous, next) but then the serializer only takes the self and overwrites it with it's own value. I'm guessing this was done on purpose as to not change our schema, I'm thinking we'll want to add it anyway as it won't break out front end and JSON API spec wants to see those fields. I'll create a new ticket for this so you don't have to change our specs and docs. For now, I think we can leave it as is so we can easily hook it up with that new ticket.

Reply:
made ticket: https://app.zenhub.com/workspaces/va-mobile-60f1a34998bc75000f2a489f/issues/gh/department-of-veterans-affairs/va-mobile-app/8958

Reply:
My intention was to change as little as possible. The schema expected links so and since jsonapi-serializer doesn't add them for you, I created a concern. 

This is the schema I was trying to match:
https://github.com/department-of-veterans-affairs/vets-api/blob/master/spec/support/schemas/folders.json#L30

Reply:
yeah, looks like our json schema we compare against in specs is slightly different than our documentation https://department-of-veterans-affairs.github.io/va-mobile-app/api/#/paths/~1v0~1messaging~1health~1folders/get